sql/postgresql/PostgreSQLParser.g4

```diff
@@ -3127,22 +3014,18 @@ select_clause
    ;
 
 simple_select
-   :
-/*??? unable to find in doc syntax like select  into sysrec * from system where name = new.sysname; - got from plpgsql.sql test script*/
-
-   SELECT (opt_all_clause into_clause opt_target_list | distinct_clause target_list) into_clause from_clause where_clause group_clause having_clause window_clause # simple_select_select
-   | values_clause # simple_select_values
-   | TABLE relation_expr # simple_select_table
-/*this replacement solves mutual left-recursive problem betwee simple_select and select_clause */
-
-   | simple_select set_operator_with_all_or_distinct (simple_select | select_with_parens) # simple_select_union_except_intersect
-   | select_with_parens set_operator_with_all_or_distinct (simple_select | select_with_parens) # simple_select_union_except_intersect
-   //this is replaced by rules above ^^^
-
-   //| select_clause UNION all_or_distinct select_clause | select_clause INTERSECT all_or_distinct
-
-   // select_clause | select_clause EXCEPT all_or_distinct select_clause
-
+   : ( SELECT (opt_all_clause into_clause opt_target_list | distinct_clause target_list)
+           into_clause
+           from_clause
+           where_clause
+           group_clause
+           having_clause
+           window_clause
+       | values_clause
+       | TABLE relation_expr
+       | select_with_parens set_operator_with_all_or_distinct (simple_select | select_with_parens)
+     )
+        (set_operator_with_all_or_distinct (simple_select | select_with_parens))*
    ;
 
 set_operator
```

Comment:
It seems PG treats INTERSECT tighter than UNION and EXCEPT but in our grammar file it's the same?

Reply:
https://www.postgresql.org/docs/current/queries-union.html

As shown here, you can use parentheses to control the order of evaluation. Without parentheses, UNION and EXCEPT associate left-to-right, but INTERSECT binds more tightly than those two operators. Thus